### ISSUE TITLE
Web UI `orderBy` automatically removes limit of 40 per page

### ISSUE THREAD

**Issue by chrisvariety** [rethinkdb/rethinkdb#3912]
Super simple query like `r.db('ti').table('analyticsPurchases').orderBy('created')` on a table with more than 40 documents will return all the rows in the Web UI, not just 40.

Not a big deal but figured I would let y'all know. Apologies if this has already been reported.

**Comment by mglukhovsky** [comment]
Thanks, @chrisvariety! I don't think we've see this before, thanks for reporting it.

@deontologician, could you take a look at this?

**Comment by mlucy** [comment]
I think this is because `orderBy` without an index is forced to load everything into memory, and it returns an array to indicate this.  If you do `r.table('analyticsPurchases').indexCreate('created'); r.table('analyticsPurchases').orderBy({index: 'created'})` then RethinkDB can efficiently do the ordering without loading everything off disk, which means it will return a stream, which should display only 40 documents like normal.

**Comment by deontologician** [comment]
Yeah, but the webui should probably still page the results. I'll check it out wednesday

**Comment by neumino** [comment]
That's a regression. The data explore use to trunk results even when an Array is returned. It's relatively easy to do. You just need to use the same code path for Arrays and Cursors (since Arrays implements the Cursors interface).

**Comment by danielmewes** [comment]
As @marshall007 mentioned in https://github.com/rethinkdb/rethinkdb/issues/4760 , it can be useful to have the full query result for copy&paste / export purposes.

When we fix this, we should make sure that the "Raw view" still displays the full result set.

**Comment by deontologician** [comment]
does that mean we should try to exhaust the cursor when they switch to the raw view tab?

**Comment by chrisvariety** [comment]
"Raw View" doesn't currently display all results, so seems kind of separate from this issue.

I think you're going to run into problems trying to exhaust a cursor on tables returning a large amount of rows -- Chrome hangs pretty bad at 12000 rows.

I currently do this silly thing to get all the results for a query: `r.expr({all: r.table('foo').coerceTo('array')})` so I agree it would be nice for a more official way to do this, but not sure how much that is related to this issue.

**Comment by marshall007** [comment]
@deontologician @chrisvariety to clarify, in #4760 I was proposing that the "display all results in Raw view" behavior only apply to arrays.  Basically "Tree view" should always be paged (current behavior is broken for arrays) and "Raw view" should only be paged for streams (current behavior is "correct").

**Comment by danielmewes** [comment]
@deontologician No, I meant what @marshall007 just said.

**Comment by danielmewes** [comment]
See https://github.com/rethinkdb/rethinkdb/issues/2202 for an option to download data directly from the web UI.
